affected_purls:
- pkg:deb/debian/python3-typed-ast@1.3.1-1
- pkg:pypi/typed-ast@1.3.0
- pkg:pypi/typed-ast@1.3.1
aliases:
- CVE-2019-19275
- GHSA-7xxv-wpxj-mx5v
fixed_purl:
- pkg:deb/debian/python3-typed-ast@1.4.2-1
- pkg:pypi/typed-ast@1.3.2
references:
- id: 149049
  reference_id: ''
  url: https://bugs.python.org/issue36495
- id: 149045
  reference_id: ''
  url: https://github.com/python/cpython/commit/a4d78362397fc3bced6ea80fbc7b5f4827aec55e
- id: 149046
  reference_id: ''
  url: https://github.com/python/cpython/commit/dcfcd146f8e6fc5c2fc16a4c192a0c5f5ca8c53c
- id: 149047
  reference_id: ''
  url: https://github.com/python/typed_ast/commit/156afcb26c198e162504a57caddfe0acd9ed7dce
- id: 149048
  reference_id: ''
  url: https://github.com/python/typed_ast/commit/dc317ac9cff859aa84eeabe03fb5004982545b3b
- id: 149050
  reference_id: ''
  url: https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/LG5H4Q6LFVRX7SFXLBEJMNQFI4T5SCEA/
- id: 149052
  reference_id: CVE-2019-19275
  url: https://nvd.nist.gov/vuln/detail/CVE-2019-19275
- id: 149053
  reference_id: GHSA-7xxv-wpxj-mx5v
  url: https://github.com/advisories/GHSA-7xxv-wpxj-mx5v
severities:
- id: 21025
  reference_id: 149053
  scoring_elements: ''
  scoring_system: cvssv3.1_qr
  value: HIGH
summary: typed_ast 1.3.0 and 1.3.1 has an ast_for_arguments out-of-bounds read. An
  attacker with the ability to cause a Python interpreter to parse Python source (but
  not necessarily execute it) may be able to crash the interpreter process. This could
  be a concern, for example, in a web-based service that parses (but does not execute)
  Python code. (This issue also affected certain Python 3.8.0-alpha prereleases.)
vulnerability_id: VCID-j86z-251z-aaaf
weaknesses: []
